PactNet.Tests/PactUriOptionsTests.cs

```diff
@@ -5,31 +5,93 @@ namespace PactNet.Tests
 {
     public class PactUriOptionsTests
     {
+
+        // This is an experimental test set.
+        // Use of Data Driving improves readability and 
+        // shows all the permutations run by one test
+        [Theory]
+        [InlineData( ""        , ""            , "username is null or empty." ) ] 
+        [InlineData( ""        , "somepassword", "username is null or empty." ) ] 
+        [InlineData( "goodname", ""            , "password is null or empty." ) ] 
+        [InlineData( null      , ""            , "username is null or empty." ) ]
+        [InlineData( ""        , null          , "username is null or empty." ) ]
+        [InlineData( null      , null          , "username is null or empty." ) ]
+        [InlineData( "bad:name", ""            , "username contains a ':' character, which is not allowed." ) ] // RFC 2617 compliance
+        [InlineData( "bad:name", "goodpass"    , "username contains a ':' character, which is not allowed." ) ] // RFC 2617 compliance
+        [InlineData( "bad:name", null          , "username contains a ':' character, which is not allowed." ) ] // RFC 2617 compliance
+        [InlineData( "goodname", null          , "password is null or empty." ) ] // RFC 2617 compliance
+        [InlineData( "goodname", ""            , "password is null or empty." ) ] // RFC 2617 compliance
+        public void Ctor_takesInvalidUserNameandPasswordParamsThatThrowArgException(string username, string password, string expectedMessage)
+        {
+            Exception e = Assert.Throws <ArgumentException> (() => new PactUriOptions(username, password));
+            Assert.Equal(expectedMessage , e.Message);
+        }
+
+        [Theory]
+        [InlineData("some@user","password")]
+        [InlineData("someuser","password")]
+        [InlineData("someuser","pass word")]
+        public void Ctor_AllowUserNamesAndPasswords(string username, string password )

```

Comment:
Maybe I am missing something, but I do not see a need for these three cases, one would be enough. All negative cases you have consider above. So for me your test cases look like testing yet another set of two numbers to verify if my function adding numbers correctly. Furthermore I think the most important thing here is that your code will not throw exception. I will probably suggest change this test to something more like:

```csharp
        [Fact]
        public void Ctror_WhenAcceptableUserNameAndPasswordProvided_ShouldNotThrowEception()
        {
            Assert.Null(Record.Exception(() => new PactUriOptions("dummyUserName", "dummyPassword")));
        }
```

Even if you do not like asserting on exception I would recommend to check just single case e.g. "user","password", unless you have a git reason to add yet another one.

Reply:
the name with an @ it failed (turned to be in ruby) so its a motivator to check compliance against RFC 2716 (URLs with username/passwords in them)

Reply:
Ok, but is it possible that this test will failed due to ruby implementation? `PactUriOptions` looks like well isolated class, which do not have nothing to do with a ruby code. 

I will suggest to remove this test case, because there `@` is not related to this class code. Maybe if you are very keen you can add some integration test class, which `truly` will test that. Anyway, decision is yours.